admin/automation/sync.py

```diff
@@ -0,0 +1,231 @@
+            "-n",
+            "--dry",
+            action="store_true",
+            help="Print information without creating anything",
+        )
+        self.parser.add_argument(
+            "-d",
+            "--debug",
+            action="store_true",
+            help="Print debugging information",
+        )
+        self.parser.add_argument(
+            "-p",
+            "--projects",
+            help="Number of projects to process",
+            default=1,
+            type=int,
+        )
+        self.parser.add_argument(
+            "--repo",
+            help="Repository where the automation should happen",
+            default="ngipkgs",
+        )
+        self.parser.add_argument(
+            "--credentials",
+            help="Directory from which GitHub credentials are loaded",
+            type=dir_path,
+            default="./.env",
+        )
+        self.parser.add_argument(
+            "--template",
+            help="Directory for the project template",
+            type=dir_path,
+            default="./template",
+        )
+
+        self.args = self.parser.parse_args()
+
+
+def dir_path(string):
+    if os.path.isdir(string):
+        return string
+    else:
+        raise NotADirectoryError(string)
+
+
+def load_credentials(directory):
+    for filename in os.listdir(directory):
+        file_path = os.path.join(directory, filename)
+
+        if os.path.isfile(file_path):
+            with open(file_path, "r") as file:
+                content = file.read().strip()
+                os.environ[filename] = content
+
+
+class Deliverable(str, Enum):
+    SERVICES = "services"
+    EXECUTABLES = "executables"
+    LIBRARIES = "libraries"
+    TESTS = "tests"
+    DEVENV = "development environment"
+
+
+@dataclass
+class Project:
+    name: str
+    branch_name: str = field(init=False)
+    description: str = "\n### Websites"
+    websites: Series = field(default_factory=Series)
+
+    def __post_init__(self):
+        self.branch_name = f"projects/{self.name}"
+
+    def append_websites(self, websites: List[str]):
+        self.websites = pd.concat(
+            [self.websites, pd.Series(websites)], ignore_index=True
+        )
+
+    # TODO: make a template for this
+    def update_description(self):
+        if len(self.websites) > 0:
+            pass
+
+        for site in self.websites:
+            self.description += f"\n- {site}"
+
+    def clean_websites(self):
+        """Remove empty values, clean up URLs and remove duplicates"""
+        self.websites = cleanup_empty(self.websites)
+        self.websites = cleanup_urls(self.websites)
+        self.websites = self.websites.drop_duplicates()
+
+
+class Subgrants(BaseModel):
+    subgrants: List[Subgrant] = []
+
+    def get_websites(self, name: str) -> List[str]:
+        for s in self.subgrants:
+            if name == s.name:
+                return s.websites
+        return []
+
+
+class NotionProject(BaseModel):
```

Comment:
Non-blocking note for the future: We may want to do this in a separate module, since we may want to export our Notion model for all kinds of purposes -- hiding that logic away in this very particular script will be in the way for that.

Reply:
I've moved the dashboard and notion data models to separate modules so they're not hardcoded in the scripts and can be changed in the future.